ls1intum/Artemis
# `Development`: Provide web-based problem statement and feedback view to native clients.

### Checklist
#### General


#### Client

### Motivation and Context


The problem statement and the feedback view are currently to complex to move them to the native clients with a native implementation. Therefore, we want to embed them using web views.
### Description

We refactor the problem statement to be able to exist on its own. Furthermore, we add a new component that can show the feedback component based on query args.

### Steps for Testing
0. Deploy to TestServer2 (since the id's below are only valid for testServer 2, otherwise you can use another test server as well but research the exerciseID , participationID and resultId yourself)
1. Login with Artemis_admin account
2. Go to https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/problem-statement/3031
3. Verify that you can see the problem statement end it looks fine (only works for light mode, dark mode is handled in native clients (background is transparent)
4. Go to https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/feedback/3031/3931/true
5. Verify that you can see the feedback to your result (only works for light mode, dark mode is handled in native clients (background is transparent)
6. Also verify that https://artemis-test2.artemis.in.tum.de/courses/567/exercises/1513/ still looks like it should

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| problem-statement.component.ts | 82% | 96% |
| standalone-feedback.component.ts | 77% | 98% |
